### PR TITLE
Make it possible to use a different S3 endpoint for ingestion

### PR DESCRIPTION
**Pull request by machine424**
Hello,

### Description

This adresses #10893, I've started a PR to get your attention and to show you that adding the feature won't require a lot of changes.

Based on the work done in https://github.com/apache/druid/pull/9375 (I know thought that PasswordProvider has been deprecated)

Let me know what you think.

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSourceConfig.java [hunk 1]
```diff
@@ -63,6 +73,7 @@ public PasswordProvider getSecretAccessKey()
     return secretAccessKey;
   }
 
+
```
Comment (yuanlihan) [inline extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSourceConfig.java:76]: Please remove this useless blank line.

Comment (machine424) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSourceConfig.java]: Thanks, but I said this is just a "draft", I'd like to initiate a discussion and see if this feature could be supported.

**Comment by machine424** [comment]
I'd like to know what you think about this.

**Comment by machine424** [comment]
I'll ping you @jihoonson as you've reviewed https://github.com/apache/druid/pull/9375, any thoughts about this? so I can put more effort making it ready.

**Comment by jihoonson** [comment]
@machine424 thanks for bringing this up. I like the goal here, will take a look at the code soon.

**Comment by maytasm** [comment]
@machine424 This approach looks reasonable to me. Do you image any other configurations that might be useful to add? I am thinking being able to configure region in the ingestionSpec might also be useful

**Pull request closed without merging**